aten/src/ATen/native/Distance.cpp

```diff
@@ -148,14 +148,14 @@ static Tensor cdist_impl(const Tensor& x1, const Tensor& x2, const double p, c10
 Tensor cdist(const Tensor& x1, const Tensor& x2, const double p, c10::optional<int64_t> compute_mode) {
   TORCH_CHECK(x1.dim() >= 2, "cdist only supports at least 2D tensors, X1 got: ", x1.dim(), "D");
   TORCH_CHECK(x2.dim() >= 2, "cdist only supports at least 2D tensors, X2 got: ", x2.dim(), "D");
-  TORCH_CHECK(x1.size(-1) == x2.size(-1), "X1 and X2 must have the same number of columns. X1: ", x1.size(-1), " X2: ", x2.size(-1));
```

Comment:
For the repro in the issue, only changes here (in cdist) are relevant. But I went ahead and SymInt'd all cdist funcs.